Modify preprocessing module for handling AF3-generated structure training

Philament1 opened the pull request:
Main code modifications:

Added id_category (with 'pdb' as default) for clean_df to consider cases where id column isn't pdb.

If alphafold=True when training, uses the lists of residues already generated, and assumes the structures have a ‘_af3’ suffix tag.

Philament1 committed:
Modify preprocessing module for handling AF3-generated structure training

kevinmicha commented:
Hi @Philament1, 

Most of the changes look very good. Would it be possible to use `_af` as before instead of `_af3` to maintain compatibility with the AlphaFold notebook in the original repository? Unless I'm missing something, this would help ensure consistency.

Philament1 committed:
Slight adjustments to preprocessing

Philament1 committed:
utils and config file changes for more flexibility

Philament1 commented:
Hi @kevinmicha,  

Here’s a summary of the changes, they should all be backwards compatible:  

- **`preprocessing/preprocessing.py`**:  
  - Changed the tag to `_af` for consistency.  
  - Added assurance for the entry identity column to be in string format.  

- **`config.py`**:  
  - Adjusted the path to the structures.  

- **`utils/biology_utils.py`**:  
  - In `check_train_test_identity()`, added a parameter `ag_diff=True`.  
    - This can be set to `False` if antigen checks are not required (for datasets where some or all antigens are identical).  

- **`utils/torch_utils.py`**:  
  - In `create_test_set()`, 
    - Added a `threshold=0.9` parameter to allow adjustment of the sequence identity check.  
    - Added the `ag_diff` parameter for consistency with `biology_utils.py`.  
  - In `training_routine()`, added the ability to select specific layers of ANTIPASTI for training via the `train_layers` parameter (default: `'all'`).  
    - Setting `train_layers='fc'` will freeze the CNN `'conv'` layers and train only the final fully connected layer.
